pkg/controller/infinispan/resources/secret/secret_controller.go

```diff
@@ -124,6 +141,99 @@ func (s secretResource) createSecret(name, label string, identities []byte) erro
 	return nil
 }
 
+func (s secretResource) reconcileKeystoreSecret() (*reconcile.Result, error) {
+	i := s.infinispan
+	if !i.IsEncryptionEnabled() {
+		return nil, nil
+	}
+
+	keystoreSecret := &corev1.Secret{}
+	if result, err := kube.LookupResource(i.GetKeystoreSecretName(), i.Namespace, keystoreSecret, s.client, s.log); result != nil {
+		return result, err
+	}
+
+	_, err := controllerutil.CreateOrUpdate(context.TODO(), s.client, keystoreSecret, func() error {
+		if ownerRef := metav1.GetControllerOf(keystoreSecret); ownerRef != nil && ownerRef.UID != s.infinispan.UID {
+			return fmt.Errorf("Unable to SetControllerReference for secret %s as it's already owned  by %s:%s:%s",
+				keystoreSecret.Name, ownerRef.Kind, ownerRef.Name, ownerRef.UID)
+		}
+		return controllerutil.SetControllerReference(i, keystoreSecret, s.scheme)
+	})
+	if err != nil {
+		return &reconcile.Result{}, err
+	}
+	return nil, nil
+}
+
+func (s secretResource) reconcileTruststoreSecret() (*reconcile.Result, error) {
+	i := s.infinispan
+	if !i.IsClientCertEnabled() {
+		return nil, nil
+	}
+
+	trustSecret := &corev1.Secret{
```

Comment:
If we need validation, it's better to validate before the structure construction.